frontend/src/app/project/logs/container-logs.component.ts

```diff
@@ -0,0 +1,154 @@
+import {AfterViewInit, Component, ElementRef, OnDestroy, OnInit, ViewChild} from "@angular/core";
+import {ActivatedRoute, ParamMap} from "@angular/router";
+import {RestService} from "../../rest/rest.service";
+import {Project} from "../project";
+import {ProjectVersion} from "../project-version";
+import {Container, PodAndContainer} from "./model";
+import {MatLegacySelectChange} from "@angular/material/legacy-select";
+import {WebSocketServiceWrapper} from "../../websocket/web-socket-service-wrapper.service";
+import {Observable, ReplaySubject, Subject, zip} from "rxjs";
+import {map, shareReplay, tap} from "rxjs/operators";
+import {AnsiUp} from "ansi_up";
+
+@Component({
+  selector: 'on-container-logs',
+  templateUrl: './container-logs.component.html',
+  styleUrls: ['./container-logs.component.scss']
+})
+export class ContainerLogsComponent implements OnInit, AfterViewInit, OnDestroy {
+
+  project: Project;
+  projectVersion: ProjectVersion;
+  podAndContainers: Array<PodAndContainer> = [];
+  selectedPod: PodAndContainer;
+  selectedContainer: Container;
+  lines: Array<string> = [];
+  lines$: Observable<Array<string>> = new ReplaySubject(1);
+  filteredLines$: Observable<Array<string>>;
+  filterString: string = '';
+  error = false;
+
+  private ansiUp = new AnsiUp();
+
+  @ViewChild('console') console: ElementRef<HTMLDivElement>;
+
+  constructor(private route: ActivatedRoute,
+              private rest: RestService,
+              private wsService: WebSocketServiceWrapper) {
+    this.ansiUp.use_classes = true;
+  }
+
+  ngOnInit() {
+    this.route.paramMap.subscribe((params: ParamMap) => {
+      let projectId = params.get('id');
+      let projectVersionId = params.get('versionId');
+      zip(this.initPodsAndContainers(projectId, projectVersionId), this.initProjectAndVersion(projectId, projectVersionId))
+        .subscribe({
+          next: () => {
+            this.updateLogSubscription();
+          },
+          error: error => {
+            this.error = true;
+          }
+        })
+    });
+    this.wsService.getLogStream().subscribe(message => {
+      const keepScrollPositionAtBottom = this.isScrolledToBottom();
+      this.lines.push(...message.lines);
+      (this.lines$ as Subject<Array<string>>).next(this.lines)
+      if (keepScrollPositionAtBottom) {
```

Comment:
That would not work because then you could not stop the auto-scrolling by scrolling up to a certain point. We need to check whether the view is scrolled down before appending the new log lines and finally adjust the scroll position, if necessary.

Reply:
ok, then it makes sense to save it to a variable before the new log lines arrive in the console.